lumicks/pylake/tests/test_imaging_mixins.py

```diff
@@ -0,0 +1,209 @@
+import pytest
+import numpy as np
+
+from lumicks.pylake.detail.imaging_mixins import TiffExport
+
+
+def tiffexport_factory(
+    images,
+    frames=True,
+    image_metadata=True,
+    timestamp_ranges=True,
+    writer_kwargs=True,
+    start_time=0,
+    exposure_time=5,
+    frame_time=10,
+):
+    def tiff_frames_factory(frames):
+        def tiff_frames(iterator=False):
+            if not iterator:
+                return frames
+
+            def frames_iterator():
+                for frame in frames:
+                    yield frame
+
+            return frames_iterator()
+
+        return tiff_frames
```

Comment:
whoa, holy indentation batman! maybe:
```suggestion
    def tiff_frames_factory(frames):
        return lambda iterator: (frame for frame in frames) if iterator else frames
```

I gave it a quick try and all tests still pass

Reply:
Thank you :-) Indeed, too many indentations!
As I really do not like lambdas, would you be satisified with:
```
def tiff_frames_factory(frames):
    def tiff_frames(iterator=False):
        return (frame for frame in frames) if iterator else frames
    return tiff_frames
```

Reply:
My vote for this one would go to lambda.

Reply:
OK, overruled. Will change it.

Reply:
yeah, in general I get your point about sometimes it's nice to have a named function to indicate what it's actually doing. In this particular case though I think having an extra nested named function is actually more confusing. Maybe using the lambda and just adding a comment line explaining why we need either a generator or just the list/array as a compromise?